omnigan/trainer.py

```diff
@@ -372,6 +392,8 @@ def run_epoch(self):
             self.update_g(multi_domain_batch)
             if self.d_opt is not None:
                 self.update_d(multi_domain_batch)
+            if self.ADVENT_D_opt is not None:
```

Comment:
if advent_d were a substet of D we wouldn't need that extra step and we'd keep the clean code?